app/components/modal-impersonate-member.js

```diff
@@ -0,0 +1,55 @@
+import ModalComponent from 'ghost-admin/components/modal-base';
+import copyTextToClipboard from 'ghost-admin/utils/copy-text-to-clipboard';
+import {alias} from '@ember/object/computed';
+import {inject as service} from '@ember/service';
+import {task, timeout} from 'ember-concurrency';
+
+const URL_FETCH_TIMEOUT = 60000; // 1 minute timeout as token lives for 10 minutes
+
+export default ModalComponent.extend({
+    config: service(),
+    store: service(),
+
+    classNames: 'modal-impersonate-member',
+
+    signinUrl: null,
+    member: alias('model'),
+
+    init() {
+        this._super(...arguments);
+    },
+
+    didInsertElement() {
+        this._super(...arguments);
+
+        this._signinUrlUpdateTask.perform();
+    },
+
+    actions: {},
+
+    copySigninUrl: task(function* () {
```

Comment:
Even though the main function `copyTextToClipboard` is not async had to use a `task` abstraction here so that we could use different `GhTaskButton` states out of the box (for "Copy Link" -> "Copied" logic)